core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectLifecycle.scala

```diff
@@ -99,4 +102,36 @@ class BpmBillingProjectCreator(billingRepository: BillingRepository,
     } catch {
       case exception: Exception => Future.failed(exception)
     }
+
+  override def preDeletionSteps(projectName: RawlsBillingProjectName, ctx: RawlsRequestContext): Future[Unit] =
+    for {
+      _ <- billingRepository.getCreationStatus(projectName).flatMap {
+        case CreationStatuses.CreatingLandingZone =>
+          Future.failed(
+            new BillingProjectDeletionException(
+              RawlsErrorReport(
+                s"Billing project ${projectName.value} cannot be deleted because its landing zone is still being created"
+              )
+            )
+          )
+        case _ => Future.successful()
```

Comment:
Am I correct that here I don't want to replace `flatMap` with `map` because I do actually want to handle all the possible creation statuses (`map` maps None -> None, but that doesn't help me here)?

Reply:
OK, I made this change, but I don't understand how it's any better.

Reply:
We talked, and I misunderstood what Blake was suggested. New code has been pushed up.